compiler/semtypes.nim

```diff
@@ -147,7 +148,11 @@ proc semEnum(c: PContext, n: PNode, prev: PType): PType =
         declarePureEnumField(c, e)
     if (let conflict = strTableInclReportConflict(symbols, e); conflict != nil):
       wrongRedefinition(c, e.info, e.name.s, conflict.info)
-    inc(counter)
+    if counter == high(typeof(counter)):
+      if i > 1 and result.n[i-2].sym.position == high(int):
```

Comment:
Why is the logic not as simple as

```suggestion
    if counter == high(typeof(counter)):
```

Reply:
It needs to take care of this case
```nim
type
  Test = enum
    A = int64.high()
    B
```
If the counter stays the same as the maximum, it won't report overflow errors for this case.